### PR TITLE
ETFE-3065: Build Review answers page. (COD-05)

### PR DESCRIPTION
**Pull request by joenjones07**

### CODE REVIEW

File: app/navigation/JourneyTypeNavigator.scala [hunk 1]
```diff
@@ -55,6 +56,13 @@ class JourneyTypeNavigator @Inject()() extends BaseNavigator {
     case CheckYourAnswersJourneyTypePage => (userAnswers: UserAnswers) =>
       routes.DraftMovementController.onPageLoad(userAnswers.ern, userAnswers.arc)
 
+    case JourneyTypeReviewPage => (userAnswers: UserAnswers) =>
```
Comment (machinesleet) [inline app/navigation/JourneyTypeNavigator.scala:59]: Not related to your change, but you might want to add `//noinspection ScalaStyle - Cyclomatic Complexity` here

File: test/controllers/sections/journeyType/JourneyTypeReviewControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ */
+
+package controllers.sections.journeyType
+
+import base.SpecBase
+import controllers.actions.{FakeDataRetrievalAction, FakeMovementAction}
+import forms.sections.journeyType.JourneyTypeReviewFormProvider
+import mocks.services.MockUserAnswersService
+import models.UserAnswers
+import models.requests.DataRequest
```
Comment (machinesleet) [inline test/controllers/sections/journeyType/JourneyTypeReviewControllerSpec.scala:24]: can be removed

File: test/controllers/sections/journeyType/JourneyTypeReviewControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import models.sections.ReviewAnswer
+import models.sections.ReviewAnswer.KeepAnswers
+import navigation.FakeNavigators.FakeJourneyTypeNavigator
+import pages.sections.journeyType.JourneyTypeReviewPage
+import play.api.data.Form
+import play.api.i18n.Messages
```
Comment (machinesleet) [inline test/controllers/sections/journeyType/JourneyTypeReviewControllerSpec.scala:30]: can be removed

File: app/navigation/JourneyTypeNavigator.scala [hunk 4]
```diff
@@ -55,6 +56,13 @@ class JourneyTypeNavigator @Inject()() extends BaseNavigator {
...
+      userAnswers.get(JourneyTypeReviewPage) match {
```
Comment (machinesleet) [inline app/navigation/JourneyTypeNavigator.scala:60]: Should there be a test for these conditions in the JourneyTypeNavigatorSpec?

File: test/mocks/viewmodels/MockCheckYourAnswersJourneyTypeHelper.scala [hunk 5]
```diff
@@ -17,19 +17,19 @@
 package mocks.viewmodels
 
 import models.requests.DataRequest
-import org.scalamock.handlers.CallHandler2
+import org.scalamock.handlers.{CallHandler2, CallHandler3}
```
Comment (machinesleet) [inline test/mocks/viewmodels/MockCheckYourAnswersJourneyTypeHelper.scala:20]: CallHandler2 can be removed

**Comment by machinesleet** [comment]
For the Summary pages where `onReviewPage` has been added, is it worth adding tests for when the actions of the `SummaryListRowViewModel` are an empty Seq?

File: app/viewmodels/checkAnswers/sections/journeyType/GiveInformationOtherTransportSummary.scala [hunk 6]
```diff
@@ -27,14 +27,14 @@ import viewmodels.implicits._
...
-  def row()(implicit request: DataRequest[_], messages: Messages): Option[SummaryListRow] =
+  def row(onReviewPage: Boolean)(implicit request: DataRequest[_], messages: Messages): Option[SummaryListRow] =
     request.userAnswers.get(GiveInformationOtherTransportPage).map {
       answer =>
 
         SummaryListRowViewModel(
           key = "giveInformationOtherTransport.checkYourAnswers.label",
           value = ValueViewModel(HtmlFormat.escape(answer).toString),
-          actions = Seq(
+          actions = if (onReviewPage == false) {Seq(
```
Comment (machinesleet) [inline app/viewmodels/checkAnswers/sections/journeyType/GiveInformationOtherTransportSummary.scala:37]: can be `if (!onReviewPage)`

File: app/viewmodels/checkAnswers/sections/journeyType/JourneyTimeDaysSummary.scala [hunk 7]
```diff
@@ -26,20 +26,20 @@ import viewmodels.implicits._
...
-  def row()(implicit request: DataRequest[_], messages: Messages): Option[SummaryListRow] = {
+  def row(onReviewPage: Boolean)(implicit request: DataRequest[_], messages: Messages): Option[SummaryListRow] = {
     val answers = request.userAnswers
     answers.get(JourneyTimeDaysPage).map {
       answer =>
         SummaryListRowViewModel(
           key = "journeyTimeDays.checkYourAnswersLabel",
           value = ValueViewModel(messages("journeyTimeDays.checkYourAnswersValue", answer.toString)),
-          actions = Seq(
+          actions = if (onReviewPage == false) {Seq(
```
Comment (machinesleet) [inline app/viewmodels/checkAnswers/sections/journeyType/JourneyTimeDaysSummary.scala:36]: can be `if (!onReviewPage)`
